pkg/services/authz/rbac/cache.go

```diff
@@ -1,29 +1,81 @@
 package rbac
 
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"time"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/infra/remotecache"
+)
+
 func userIdentifierCacheKey(namespace, userUID string) string {
-	return "UID_" + namespace + "_" + userUID
+	return namespace + ".uid_" + userUID
 }
 
 func userIdentifierCacheKeyById(namespace, ID string) string {
-	return "ID_" + namespace + "_" + ID
+	return namespace + ".id_" + ID
 }
 
 func anonymousPermCacheKey(namespace, action string) string {
-	return namespace + "_anonymous_" + action
+	return namespace + ".perm_anonymous_" + action
 }
 
 func userPermCacheKey(namespace, userUID, action string) string {
-	return namespace + "_" + userUID + "_" + action
+	return namespace + ".perm_" + userUID + "_" + action
 }
 
 func userBasicRoleCacheKey(namespace, userUID string) string {
-	return namespace + "_" + userUID
+	return namespace + ".basic_role_" + userUID
 }
 
 func userTeamCacheKey(namespace, userUID string) string {
-	return namespace + "_" + userUID
+	return namespace + ".team_" + userUID
 }
 
 func folderCacheKey(namespace string) string {
-	return namespace
+	return namespace + ".folders"
+}
+
+type cacheWrap[T any] struct {
+	cache  remotecache.CacheStorage
+	logger log.Logger
+	ttl    time.Duration
+}
+
+func newCacheWrap[T any](cache remotecache.CacheStorage, logger log.Logger, ttl time.Duration) *cacheWrap[T] {
+	return &cacheWrap[T]{cache: cache, logger: logger, ttl: ttl}
+}
+
+func (c *cacheWrap[T]) Get(ctx context.Context, key string) (T, bool) {
+	var value T
+	data, err := c.cache.Get(ctx, key)
+	if err != nil {
+		if !errors.Is(err, remotecache.ErrCacheItemNotFound) {
```

Comment:
If I use the cache from the authlib as a local cache, this is not the error that's being returned.